clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -681,6 +681,16 @@ synchronized void authenticationFailed(AuthenticationException e) {
             request.fatalError(e);
     }
 
+    synchronized void abortPendingTransactionalRequests() {
+        KafkaException shutdownException = new KafkaException("The producer closed forcefully");
```

Comment:
Just a nit, but the name of the api suggests that this could be called in other scenarios than closing. Perhaps we should rename the method to something like `abortPendingRequestsOnForceClose`. Or maybe just `close()`? 😉

Reply:
Renamed it :D